### DISCUSSION TITLE
Переделать CalendarRange/DateRangeInput

### DISCUSSION THREAD

**Discussion by Semigradsky** [VKCOM/VKUI#6894] [Ideas] [answered]
Сейчас выбор диапазона дат выглядит "очень не обычно".

Как это работает в других компонентах на примере MUI (https://mui.com/x/react-date-pickers/date-range-calendar/). Пусть изначально выбран диапазон `2024-05-06`-`2024-05-10`, мы хотим поменять диапазон на `2024-05-08`-`2024-05-12`:
1. Щёлкаем на `2024-05-08`. Компонент считает что первый выбор даты - это выбор значения `from`. В зависимости от реализации компонента, значение `to` будет или сброшено (тогда имеем диапазон `2024-05-08`-`null`), или останется предыдущим (тогда имеем диапазон `2024-05-08`-`2024-05-10`).
2. Щёлкаем на `2024-05-12`. Компонент считает что второй выбор даты - это выбор значения `to`. Мы получили новый диапазон `2024-05-08`-`2024-05-12`, который и хотели.

В компоненте VKUI же всё работает совсем по другому, какое значение мы меняем (`from` или `to`) зависит от того на какой день мы щёлкнем - в пределах ранее выбранного диапазона или нет. Кроме того, что это не очевидно, это ещё и не удобно.

В доке можете попробовать выставить начальный диапазон и посчитать сколько раз будете открывать модалку календаря, чтобы выбрать новый диапазон - https://vkcom.github.io/VKUI/#/DateRangeInput. Если выбрана опция "Закрывать календарь послы выбора даты", то будете открывать модалку минимум 3 раза (если повезёт или уже знаете в каком порядке надо щёлкать по дням). Но в идеале должно быть только одно открытие модалки, как в других Date Range Pickerах.

cc @mendrew

**Comment by andrey-medvedev-vk** [comment, marked as answer]
Переделали логику выбора диапазона в [v7.0.0](https://github.com/VKCOM/VKUI/releases/tag/v7.0.0)
issue: https://github.com/VKCOM/VKUI/issues/7670
PR: https://github.com/VKCOM/VKUI/pull/7682

Максимально её упростили.

Для выбора диапазона нужно проставить первую и вторую дату. Третий клик по дате сбрасывает выбранный диапазон и вновь проставляет первую дату нового диапазона.
Детали в [issue](https://github.com/VKCOM/VKUI/issues/7670) и [PR](https://github.com/VKCOM/VKUI/pull/7682).

https://github.com/user-attachments/assets/096a4db5-9cf2-4e34-ac9d-7909046ee223
